Documentation/network/bgp-control-plane/bgp-control-plane-v2.rst

```diff
@@ -124,11 +124,14 @@ Transport
 ---------
 
 The transport section of ``CiliumBGPPeerConfig`` can be used to configure custom source and
-destination ports for the BGP session with the peer. ``LocalPort`` and ``PeerPort`` fields are used to
-configure the source and destination ports, respectively.
+destination ports for the BGP session with the peer. The ``localPort`` and ``peerPort`` fields
```

Comment:
Did you have customers ask for exposing `localPort`?

I was going to drop it originally from the example.  Glad you suggested that.  It may be better to remove all references to `localPort` from the docs.

I had only tested "clear ip bgp neighbor X", but now that I think of it, we may see the same delay when the remote router restarts.  Graceful restart may help, but if the remote side crashes without sending a notification, we may see the same delay while waiting for the socket to time out.

Reply:
> Did you have customers ask for exposing localPort?

No, this was mostly done to expose some of the GoBGP transport options. Thinking more about it, it might have been unnecessary and dangerous. I agree. We can remove the reference from the docs and leave in a warning, and subsequently remove this option altogether from the CRD. 

cc @rastislavs  @YutaroHayakawa  thoughts ?

Reply:
I think in BGPv1 this was exposed via an annotation, so there probably was a use-case for exposing it (that hard way)?

Anyway, I guess the only use-case would be when Cilium cannot initiate connections (e.g. because of firewall restrictions) and at the same time the default port can not be used for binding (e.g. in case of more virtual router instances on Cilium, or if another BGP daemon is running on the Cilium node)?

I would vote from removing it from docs, and keeping it in CRD - with an additional note on implications when configuring it.

Reply:
> I think in BGPv1 this was exposed via an annotation, so there probably was a use-case for exposing it (that hard way)?

No, it is not the same listen port as in BGPv1. That listen port  (with annotation) is for gobgp to accept incoming connections only on that specific port. This local/remote port is used when gobgp tries to initiate a TCP connection.

Reply:
We can keep it in CRD and document its behaviour so we do not make API change.

Reply:
> This local/remote port is used when gobgp tries to initiate a TCP connection and is per peer configuration.

Ah sorry, so if it means the source port when originating a connection to a specific peer, then I agree that exposing it may cause more troubles than benefit.